packages/lix-sdk/src/observe/determine-schema-keys.test.ts

```diff
@@ -0,0 +1,291 @@
+	
+	// Should include schema keys for both active_version and version tables
+	expect(schemaKeys).toContain("lix_version");
+	// Note: active_version might not have a direct schema key mapping
+	
+	await lix.close();
+});
+
+test("should extract schema keys from key_value queries", async () => {
+	const lix = await openLix({});
+	
+	// Test selectCurrentLixName query from md-app
+	const currentLixNameQuery = lix.db
+		.selectFrom("key_value")
+		.where("key", "=", "lix_name")
+		.select("value");
+	
+	const schemaKeys = determineSchemaKeys(currentLixNameQuery.compile());
+	
+	expect(schemaKeys).toContain("lix_key_value");
+	
+	await lix.close();
+});
+
+test("should extract schema keys from state table queries", async () => {
+	const lix = await openLix({});
+	
+	// Test selectMdAstRoot query from md-app (this uses "state" table)
+	const mdAstRootQuery = lix.db
+		.selectFrom("state")
+		.where("schema_key", "=", "markdown_root_v1")
+		.where("file_id", "=", "test_file_id")
+		.selectAll();
+	
+	const schemaKeys = determineSchemaKeys(mdAstRootQuery.compile());
+	
+	// "state" is a virtual table that should map to appropriate schema keys
+	// For now, this test documents the current behavior
+	expect(Array.isArray(schemaKeys)).toBe(true);
+	
+	await lix.close();
+});
+
+test("should extract schema keys from complex join queries", async () => {
+	const lix = await openLix({});
+	
+	// Test selectCheckpoints query from md-app (complex with multiple joins)
+	const checkpointsQuery = lix.db
+		.selectFrom("change_set")
+		.leftJoin("change_set_element", "change_set.id", "change_set_element.change_set_id")
+		.leftJoin("change_set_label", "change_set.id", "change_set_label.change_set_id")
+		.leftJoin("label", "label.id", "change_set_label.label_id")
+		.where("label.name", "=", "checkpoint")
+		.selectAll("change_set");
+	
+	const schemaKeys = determineSchemaKeys(checkpointsQuery.compile());
+	
+	// Should include schema keys for all joined tables
+	expect(schemaKeys).toContain("lix_change_set");
+	expect(schemaKeys).toContain("lix_change_set_element");
+	expect(schemaKeys).toContain("lix_change_set_label");
+	expect(schemaKeys).toContain("lix_label");
+	
+	await lix.close();
+});
+
+test("should extract schema keys from subquery queries", async () => {
+	const lix = await openLix({});
+	
+	// Test selectActiveFile query from md-app (has subquery)
+	const activeFileQuery = lix.db
+		.selectFrom("file")
+		.where(
+			"id",
+			"=",
+			lix.db
+				.selectFrom("key_value")
+				.where("key", "=", "flashtype_active_file")
+				.select("value")
+		)
+		.select(["id", "path", "metadata"]);
+	
+	const schemaKeys = determineSchemaKeys(activeFileQuery.compile());
+	
+	// Should include schema keys for both main query and subquery tables
+	expect(schemaKeys).toContain("lix_file");
+	expect(schemaKeys).toContain("lix_key_value");
+	
+	await lix.close();
+});
+
+test("should handle queries with aliases", async () => {
```

Comment:
I would add an aliases join with a schema key like `lix_key_value`. so just add a innerJoin(key_value) and expect key value to exsits too